src/main/java/com/radixdlt/atommodel/Atom.java

```diff
@@ -413,9 +358,34 @@ private void setJsonSignatures(Map<String, ECDSASignature> sigs) {
 		}
 	}
 
+	public static AID aidOf(Atom atom, Hasher hasher) {
+		HashCode hash = hasher.hash(atom);
+		return AID.from(hash.asBytes());
+	}
+
 	@Override
 	public String toString() {
 		String particleGroupsStr = this.particleGroups.stream().map(ParticleGroup::toString).collect(Collectors.joining(","));
-		return String.format("%s[%s:%s]", getClass().getSimpleName(), getAID(), particleGroupsStr);
+		return String.format("%s[%s]", getClass().getSimpleName(), particleGroupsStr);
+	}
+
+	@Override
+	public boolean equals(Object o) {
+		if (this == o) {
+			return true;
+		}
+		if (o == null || getClass() != o.getClass()) {
```

Comment:
Would prefer to see `instanceof` tests here, rather than getClass() equality for final classes.